ksqldb-rest-app/src/main/java/io/confluent/ksql/api/spi/EndpointResponse.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.spi;
+
+public interface EndpointResponse {
+
+  int getStatusCode();
+
+  String getStatusMessage();
+
+  Object getResponseBody();
+
+  static EndpointResponse create(final int errorCode, final String statusMessage,
+      final Object responseBody) {
```

Comment:
nit: validate params that will be stored in object state; ensuring object does not get into an invalid state.

Reply:
ack